providers/aws-ssm/provider.go

```diff
@@ -0,0 +1,53 @@
+package awsssm
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/open-feature/go-sdk/openfeature"
+)
+
+type Provider struct {
+	svc *awsService
+}
+
+type ProviderOption func(*Provider)
+
+
+func NewProvider(cfg aws.Config, opts ...ProviderOption) (*Provider, error) {
```

Comment:
There is still no way for a user to set `WithDecryption`. A provider option should be implemented to allow this.